com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/helpers/RateLimiterProvider.java

```diff
@@ -0,0 +1,28 @@
+package com.sap.cloud.lm.sl.cf.web.helpers;
+
+import io.github.resilience4j.ratelimiter.RateLimiter;
+import io.github.resilience4j.ratelimiter.RateLimiterConfig;
+
+import javax.inject.Named;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Named
+public class RateLimiterProvider {
+
+    private Map<String, RateLimiter> rateLimiters = new ConcurrentHashMap<>();
+
+    public RateLimiter getRateLimiter(String ipAddress) {
+        return rateLimiters.computeIfAbsent(ipAddress, k -> createRateLimiter());
+    }
+
+    private RateLimiter createRateLimiter() {
+        RateLimiterConfig config = RateLimiterConfig.custom()
+                                                    .limitForPeriod(5000)
+                                                    .limitRefreshPeriod(Duration.ofHours(1))
+                                                    .timeoutDuration(Duration.ZERO)
+                                                    .build();
+        return RateLimiter.of("MTA Rate Limiter", config);
```

Comment:
Would it be nicer if we create the AtomicRateLimiter explicitly?
What would happen if they change the implementation of the "of" method to return SemaphoreBasedRateLimiter in future - the cast to the AtomicRateLimiterMetrics would be broken?